cpp/inc/bond/ext/grpc/wait_callback.h

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include <bond/core/bonded.h>
+#include <grpc++/impl/codegen/status.h>
+
+#include <boost/optional.hpp>
+
+#include <chrono>
+#include <condition_variable>
+#include <functional>
+#include <memory>
+#include <mutex>
+
+namespace bond { namespace ext { namespace gRPC {
+
+/// @brief A callback type that can be manually waited upon.
```

Comment:
is it worthwhile to mention something regarding 'sync' or 'sync-ifying' the rpc calls?